pkg/pgmodel/querier/query_builder.go

```diff
@@ -427,6 +427,30 @@ func calledByTimestamp(path []parser.Node) bool {
 }
 
 var vectorSelectorExtensionRange = semver.MustParseRange(">= 0.1.3-beta")
+var rateIncreaseExtensionRange = semver.MustParseRange(">= 0.1.3-beta")
+
+func callAggregator(hints *storage.SelectHints, funcName string) *aggregators {
+	queryStart := hints.Start + hints.Range
+	queryEnd := hints.End
+	stepDuration := time.Second
+	rangeDuration := time.Duration(hints.Range) * time.Millisecond
+
+	if hints.Step > 0 {
+		stepDuration = time.Duration(hints.Step) * time.Millisecond
+	} else {
+		if queryStart != queryEnd {
+			panic("query start should equal query end")
```

Comment:
Should we be panicking here? This will kill Promscale just because hints details were unexpected.